shinylive/_url.py

```diff
@@ -0,0 +1,217 @@
+from __future__ import annotations
+
+import base64
+import json
+import os
+import re
+from pathlib import Path
+from typing import List, Literal, Optional, TypedDict, cast
+
+
+class FileContentJson(TypedDict):
+    name: str
+    content: str
+    type: Literal["text", "binary"]
```

Comment:
Thanks for the background and the links to the original typescript types. I've updated the imports statements and generally improved how FileContentJson objects are created to avoid adding the `type` field for text files.

Reply:
Thanks for the background and the links to the original typescript types. I've updated the imports statements and generally improved how FileContentJson objects are created to avoid adding the `type` field for text files.

Reply:
I added `typing_extensions` to `install_requires` but we're now running into failing checks similar to the ones that tripped us up with `lzstring`. Any ideas?